pkg/webhook/resources/setting/validator.go

```diff
@@ -1063,53 +1064,75 @@ func (v *settingValidator) validateNetworkHelper(value string) (*networkutil.Con
 		return nil, err
 	}
 
+	if name == settings.StorageNetworkName {
+		if err := v.checkStorageNetworkRangeValid(&config); err != nil {
+			return nil, err
+		}
+	} else if name == settings.VMMigrationNetworkSettingName {
+		if err := v.checkVMMigrationNetworkRangeValid(&config); err != nil {
+			return nil, err
+		}
+	}
```

Comment:
Here’s a suggestion to improve flexibility:

```suggestion
	// Add a map of setting names to their specific network range validation callbacks
	networkRangeValidators := map[string]func(*networkutil.Config) error{
		settings.StorageNetworkName:            v.checkStorageNetworkRangeValid,
		settings.VMMigrationNetworkSettingName: v.checkVMMigrationNetworkRangeValid,
	}

	if validator, ok := networkRangeValidators[name]; ok {
		if err := validator(&config); err != nil {
			return nil, err
		}
	}
```